### PR TITLE
Put together an Aff 4.0 package set

### PR DESCRIPTION
**Pull request by justinwoo**
Putting this together to test it out, hopefully I can use it as a long-living todolist to try to get the removed packages fixed.

### CODE REVIEW

**Comment by justinwoo** [comment]
Verified the set locally, let's see what Travis has to say about it...

**Comment by justinwoo** [comment]
Reminder to self that io and pipes are updated, so I need to look into those eventually

**Comment by paf31** [comment]
Thermite is updated too.

File: packages.json [hunk 1]
```diff
@@ -321,8 +345,8 @@
        "aff",
        "transformers"
     ],
-    "repo": "https://github.com/cdepillabout/purescript-email-validate.git",
-    "version": "v1.0.7"
+    "repo": "https://github.com/justinwoo/purescript-email-validate.git",
+    "version": "aff-4.0"
```
Comment (justinwoo) [inline packages.json]: NOTE TO SELF THAT THIS NEEDS TO BE UPDATED

Comment (justinwoo) [reply packages.json]: https://github.com/cdepillabout/purescript-email-validate/pull/5

File: packages.json [hunk 2]
```diff
@@ -795,16 +788,6 @@
     "repo": "https://github.com/paf31/purescript-memoize.git",
     "version": "v4.0.1"
   },
-  "milkis": {
```
Comment (justinwoo) [inline packages.json]: depends on aff promise

File: packages.json [hunk 3]
```diff
@@ -1,40 +1,53 @@
...
   },
   "aff-coroutines": {
     "dependencies": [
       "aff",
       "coroutines"
     ],
     "repo": "https://github.com/purescript-contrib/purescript-aff-coroutines.git",
-    "version": "v5.0.0"
-  },
-  "aff-promise": {
```
Comment (justinwoo) [inline packages.json]: depends on PR stuff

File: packages.json [hunk 4]
```diff
@@ -570,52 +594,6 @@
     "repo": "https://github.com/purescript/purescript-graphs.git",
     "version": "v3.0.0"
   },
-  "halogen": {
-    "dependencies": [
```
Comment (justinwoo) [inline packages.json]: halogen family wip

File: packages.json [hunk 5]
```diff
@@ -878,24 +861,6 @@
     "repo": "https://github.com/purescript-node/purescript-node-path.git",
     "version": "v2.0.0"
   },
-  "node-postgres": {
```
Comment (justinwoo) [inline packages.json]: wip

File: packages.json [hunk 6]
```diff
@@ -1106,17 +1077,6 @@
     "repo": "https://github.com/purescript/purescript-psci-support.git",
     "version": "v3.0.0"
   },
-  "pux": {
```
Comment (justinwoo) [inline packages.json]: wip

File: packages.json [hunk 7]
```diff
@@ -1244,14 +1194,6 @@
     "repo": "https://github.com/purescript/purescript-semirings.git",
     "version": "v4.0.0"
   },
-  "servant-support": {
```
Comment (justinwoo) [inline packages.json]: wip

File: packages.json [hunk 8]
```diff
@@ -1301,23 +1243,19 @@
...
+      "transformers",
+      "monoid",
+      "foldable-traversable",
+      "pipes",
+      "ansi"
     ],
     "repo": "https://github.com/owickstrom/purescript-spec.git",
-    "version": "v1.0.0"
-  },
-  "spec-quickcheck": {
```
Comment (justinwoo) [inline packages.json]: wip

File: packages.json [hunk 9]
```diff
@@ -1082,380 +992,362 @@
...
-      "maps"
+      "tuples",
+      "free",
+      "transformers",
+      "bifunctors"
     ],
     "repo": "https://github.com/bodil/purescript-smolder.git",
-    "version": "v9.0.0"
-  },
-  "sparkle": {
```
Comment (justinwoo) [inline packages.json]: I don't know how to upgrade this package from Monoid MarkupM to non-monoid so I'm just going to drop this

File: packages.json (hunk 6 above)
Comment (justinwoo) [reply packages.json]: in the end, i couldn't figure out how to make pux actually build correctly in CI. need to experiment further later.

**Comment by justinwoo** [comment]
I think the packages dropped from the set will take a while before they can be updated for use, but the other ones all work fine. Could we merge and tag this, if there aren't any problems?

**Comment by paf31** [comment]
I'm not sure if we want to drop packages between major releases like this, especially something like Halogen.

Could you please make a list of the dropped packages?

File: packages.json [hunk 10]
```diff
@@ -1109,81 +1279,111 @@
...
+      "profunctor",
+      "react",
+      "globals",
+      "dom",
+      "smolder",
+      "css"
     ],
     "repo": "https://github.com/alexmingoia/purescript-pux.git",
-    "version": "v10.0.0"
+    "version": "v12.0.0"
```
Comment (justinwoo) [inline packages.json]: This had better work, it worked here: https://travis-ci.org/justinwoo/package-sets/builds/286428685

**Comment by justinwoo** [comment]
Here are the packages that are missing:

* halogen
  WIP PR
* halogen-bootstrap
  depends on Halogen
* halogen-css
  depends on Halogen
* halogen-echarts
  depends on Halogen
* halogen-vdom
  depends on Halogen
* node-postgres
  WIP PR
* sparkle
  No PR
* spec-quickcheck
  WIP PR

There's also something seriously wrong with the travis setup here or something, as I can't get Pux to verify even though it works already for me: https://travis-ci.org/justinwoo/package-sets/builds/286428685, https://travis-ci.org/justinwoo/package-sets/builds/286432483

**Comment by justinwoo** [comment]
I guess at minimum we should wait until Halogen is in, and then the others seem a bit optional whether or not people have actually use them, but maybe Node-Postgres is important to have too.

File: packages.json (hunk 5 above)
Comment (anttih) [reply packages.json]: https://github.com/epost/purescript-node-postgres/releases/tag/v4.0.0

**Comment by paf31** [comment]
I've reset the cache and restarted the build.

**Comment by paf31** [comment]
I'm thinking it would be good to merge this into the 0.11.7 package set, since I'm thinking of making a 0.11.7 release this week to put out some bug fixes and IDE improvements.

**Comment by paf31** [comment]
Thanks very much!

**Pull request merged**